File: app/src/main/java/com/denchic45/widget/calendar/WeekCalendarListener.java
package com.denchic45.widget.calendar;

import com.denchic45.widget.calendar.model.WeekItem;

import java.time.LocalDate;

public interface WeekCalendarListener {

    void onDaySelect(LocalDate date);

    void onWeekSelect(WeekItem weekItem);

    interface OnLoadListener {

        void onWeekLoad(WeekItem weekItem);
    }
}

File: app/common/src/androidMain/kotlin/com/denchic45/studiversity/widget/calendar/WeekCalendarListener.kt
package com.denchic45.studiversity.widget.calendar

import java.time.LocalDate

interface WeekCalendarListener {
    fun onDaySelect(date: LocalDate)
    fun onWeekSelect(weekItem: com.denchic45.studiversity.widget.calendar.model.WeekItem)
    interface OnLoadListener {
        fun onWeekLoad(weekItem: com.denchic45.studiversity.widget.calendar.model.WeekItem)
    }
}